File: android/app/src/main/java/com/lightningkite/kwifttemplate/layouts/PreviewXml.kt
//
// PreviewXml.swift
// Created by Kwift XML Android
//
package com.lightningkite.kwifttemplate.layouts

import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwifttemplate.R

class PreviewXml {

    lateinit var viewName: TextView
    lateinit var pager: androidx.viewpager.widget.ViewPager
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.preview, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        viewName = view.findViewById<TextView>(R.id.viewName)
        pager = view.findViewById<androidx.viewpager.widget.ViewPager>(R.id.pager)
        
        
        return view
    }
}

File: ios/Kwift Template/swiftResources/layouts/PreviewXml.swift
//
// PreviewXml.swift
// Created by Kwift XML
//

import UIKit
import Kwift
import MapKit

public class PreviewXml {
    
    public unowned var xmlRoot: UIView!
    public func setup(_ dependency: ViewDependency) -> UIView {
        let view = LinearLayout(frame: .zero)
        view.orientation = .y
        view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
        view.gravity = .topLeft
        
        view.addSubview(
            UILabel(frame: .zero),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 0 + 8, left: 0 + 8, bottom: 0 + 8, right: 0 + 8),
            gravity: .topFill,
            weight: 0
        ) { view in 
            self.viewName = view
            view.font = UIFont.get(size: 16, style: [])
            view.numberOfLines = 0
            view.textColor = UIColor(argb: 0xFF222222)
            view.textAlignment = .center
            view.baselineAdjustment = .alignCenters
        }
        
        view.addSubview(
            UICollectionView(frame: .zero, collectionViewLayout: ViewPagerLayout()),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 0 + 0, left: 0 + 0, bottom: 0 + 0, right: 0 + 0),
            gravity: .topFill,
            weight: 1.0
        ) { view in 
            self.pager = view
            view.backgroundColor = ResourcesColors.white
            view.canCancelContentTouches = false
            view.showsHorizontalScrollIndicator = false
            view.backgroundColor = .clear
        }
        
        xmlRoot = view
        return view
    }
    
    public unowned var viewName: UILabel!
    public unowned var pager: UICollectionView!
    
}
